library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -483,6 +484,23 @@ libspdm_return_t libspdm_handle_large_request(
             if (spdm_response->header.param1
                 & SPDM_CHUNK_SEND_ACK_RESPONSE_ATTRIBUTE_EARLY_ERROR_DETECTED) {
 
+                spdm_error = (spdm_error_response_t *) (spdm_response + 1);
+                if (response_size < (sizeof(spdm_chunk_send_ack_response_t) +
+                                     sizeof(spdm_error_response_t))) {
+                    status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+                    break;
+                }
+                if ((spdm_error->header.spdm_version !=
+                     libspdm_get_connection_version(spdm_context)) ||
+                    (spdm_error->header.request_response_code != SPDM_ERROR)) {
+                    status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                    break;
+                }
+                if (spdm_error->header.param1 == SPDM_ERROR_CODE_LARGE_RESPONSE) {
+                    status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
```

Comment:
I think deeply, maybe it is more accurate to change `status = LIBSPDM_STATUS_INVALID_MSG_FIELD;` at the above line 500 to `status = LIBSPDM_STATUS_ERROR_PEER;`.
https://github.com/DMTF/libspdm/blob/138af46bffc55c0420ad32288848e8eeeae1e908/include/library/spdm_return_status.h#L108-L110

Reply:
changed